[BUG]: android build on expo failed

I try building android using this command after changing these 2 files

# expo command build
`eas build --platform android`

# files I changed inorder for the build to work
1-  `packages/app/app.json >> I remove (extra.eas.projectId) to generate new one`
2- `packages/app/scripts/eas-build-pre-install.sh >> I specify expo-cli version instead of lates (npm i -g expo-cli@6.0.8)`

I get this error in expo build, however building with react-native works.

```

C/C++: ninja: Entering directory `/home/expo/workingdir/build/packages/app/android/lib-keys-secret/.cxx/RelWithDebInfo/133g4p6s/arm64-v8a'
C/C++: /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:18:57: error: use of undeclared identifier 'authorizedAppSha1'
C/C++: /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:19:16: warning: implicit declaration of function 'getDecryptedKey' is invalid in C99 [-Wimplicit-function-declaration]
C/C++: /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:19:63: error: use of undeclared identifier 'liteInitGPCParams'
C/C++: /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:19:37: error: use of undeclared identifier 'liteInitGPCParams'
> Task :lib-keys-secret:buildCMakeRelWithDebInfo[arm64-v8a] FAILED
> Task :react-native-v8:copyReleaseJniLibsProjectOnly
> Task :app:bundleReleaseJsAndAssets
warning: the transform cache was reset.
> Task :react-native-v8:stripReleaseDebugSymbols
> Task :app:bundleReleaseJsAndAssets FAILED
Error: The resource `/home/expo/workingdir/build/packages/app/index.js` was not found.
    at /home/expo/workingdir/build/node_modules/metro/src/IncrementalBundler.js:297:24
    at gotStat (node:fs:2684:21)
    at FSReqCallback.oncomplete (node:fs:202:21)
info Run CLI with --verbose flag for more details.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings
1456 actionable tasks: 1456 executed
[stderr] error The resource `/home/expo/workingdir/build/packages/app/index.js` was not found.
[stderr] FAILURE: Build completed with 3 failures.
[stderr] 1: Task failed with an exception.
[stderr] -----------
[stderr] * What went wrong:
[stderr] Execution failed for task ':lib-keys-secret:buildCMakeRelWithDebInfo[arm64-v8a]'.
[stderr] > com.android.ide.common.process.ProcessException: ninja: Entering directory `/home/expo/workingdir/build/packages/app/android/lib-keys-secret/.cxx/RelWithDebInfo/133g4p6s/arm64-v8a'
[stderr]   [1/3] Building C object CMakeFiles/keys.dir/keys.c.o
[stderr]   FAILED: CMakeFiles/keys.dir/keys.c.o 
[stderr]   /home/expo/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --target=aarch64-none-linux-android23 --gcc-toolchain=/home/expo/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/linux-x86_64 --sysroot=/home/expo/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/linux-x86_64/sysroot -Dkeys_EXPORTS  -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security  -O2 -g -DNDEBUG -fPIC -fvisibility=hidden -MD -MT CMakeFiles/keys.dir/keys.c.o -MF CMakeFiles/keys.dir/keys.c.o.d -o CMakeFiles/keys.dir/keys.c.o -c /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c
[stderr]   /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:18:57: error: use of undeclared identifier 'authorizedAppSha1'
[stderr]       if (checkSecurityPermission(env, context, (char **) authorizedAppSha1, 3)) {
[stderr]                                                           ^
[stderr]   /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:19:16: warning: implicit declaration of function 'getDecryptedKey' is invalid in C99 [-Wimplicit-function-declaration]
[stderr]           return getDecryptedKey(env, liteInitGPCParams, sizeof(liteInitGPCParams));
[stderr]                  ^
[stderr]   /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:19:63: error: use of undeclared identifier 'liteInitGPCParams'
[stderr]           return getDecryptedKey(env, liteInitGPCParams, sizeof(liteInitGPCParams));
[stderr]                                                                 ^
[stderr]   /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:19:37: error: use of undeclared identifier 'liteInitGPCParams'
[stderr]           return getDecryptedKey(env, liteInitGPCParams, sizeof(liteInitGPCParams));
[stderr]                                       ^
[stderr]   1 warning and 3 errors generated.
[stderr]   [2/3] Building C object CMakeFiles/keys.dir/validation.c.o
[stderr]   ninja: build stopped: subcommand failed.
[stderr]   
[stderr]   C++ build system [build] failed while executing:
[stderr]       /home/expo/Android/Sdk/cmake/3.18.1/bin/ninja \
[stderr]         -C \
[stderr]         /home/expo/workingdir/build/packages/app/android/lib-keys-secret/.cxx/RelWithDebInfo/133g4p6s/arm64-v8a \
[stderr]         keys
[stderr]     from /home/expo/workingdir/build/packages/app/android/lib-keys-secret
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] 2: Task failed with an exception.
[stderr] -----------
[stderr] * What went wrong:
[stderr] java.lang.StackOverflowError (no error message)
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] 3: Task failed with an exception.
[stderr] -----------
[stderr] * What went wrong:
[stderr] Execution failed for task ':app:bundleReleaseJsAndAssets'.
[stderr] > Process 'command 'node'' finished with non-zero exit value 1
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] * Get more help at https://help.gradle.org
[stderr] BUILD FAILED in 5m 45s
Error: Gradle build failed with unknown error. See logs for the "Run gradlew" phase for more information.

```

> because the default entry file is not "index.js". so either specify an "ENTRY_FILE" env variable or create an index.js pointing to `__generated__/AppEntry.js` https://github.com/OneKeyHQ/app-monorepo/pull/2437/files#diff-66101a4c501549996a95c6753fbc127309cea4b2c38c7c98901a71af37b0518f

## The same error happened

`> Task :expo-av:compileReleaseLibraryResources
> Task :expo-barcode-scanner:compileReleaseLibraryResources
> Task :expo-camera:compileReleaseLibraryResources
> Task :expo-clipboard:compileReleaseLibraryResources
> Task :expo-constants:compileReleaseLibraryResources
> Task :expo-device:compileReleaseLibraryResources
> Task :react-native-v8:externalNativeBuildRelease
> Task :expo-file-system:compileReleaseLibraryResources
> Task :expo-font:compileReleaseLibraryResources
> Task :expo-haptics:compileReleaseLibraryResources
> Task :expo-image-loader:compileReleaseLibraryResources
> Task :expo-image-manipulator:compileReleaseLibraryResources
> Task :expo-image-picker:compileReleaseLibraryResources
> Task :expo-intent-launcher:compileReleaseLibraryResources
> Task :lib-keys-secret:buildCMakeRelWithDebInfo[arm64-v8a] FAILED
> Task :expo-keep-awake:compileReleaseLibraryResources
> Task :expo-linear-gradient:compileReleaseLibraryResources
> Task :app:bundleReleaseJsAndAssets
warning: the transform cache was reset.
                    Welcome to Metro!
              Fast - Scalable - Integrated
> Task :react-native-v8:mergeReleaseJniLibFolders
> Task :app:bundleReleaseJsAndAssets
info Writing bundle output to:, /home/expo/workingdir/build/packages/app/android/app/build/generated/assets/react/release/index.android.bundle
info Writing sourcemap output to:, /home/expo/workingdir/build/packages/app/android/app/build/generated/sourcemaps/react/release/index.android.bundle.map
info Done writing bundle output
info Done writing sourcemap output
info Copying 175 asset files
info Done copying assets
[stderr] FAILURE: Build completed with 2 failures.
[stderr] 1:
[stderr] Task failed with an exception.
[stderr] -----------
[stderr] * What went wrong:
[stderr] Execution failed for task ':lib-keys-secret:buildCMakeRelWithDebInfo[arm64-v8a]'.
[stderr] >
[stderr] com.android.ide.common.process.ProcessException: ninja: Entering directory `/home/expo/workingdir/build/packages/app/android/lib-keys-secret/.cxx/RelWithDebInfo/133g4p6s/arm64-v8a'
[stderr] [1/3] Building C object CMakeFiles/keys.dir/keys.c.o
[stderr] FAILED: CMakeFiles/keys.dir/keys.c.o
[stderr] /home/expo/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --target=aarch64-none-linux-android23 --gcc-toolchain=/home/expo/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/linux-x86_64 --sysroot=/home/expo/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/linux-x86_64/sysroot -Dkeys_EXPORTS  -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security  -O2 -g -DNDEBUG -fPIC -fvisibility=hidden -MD -MT CMakeFiles/keys.dir/keys.c.o -MF CMakeFiles/keys.dir/keys.c.o.d -o CMakeFiles/keys.dir/keys.c.o -c /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c
[stderr] /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:18:57: error: use of undeclared identifier 'authorizedAppSha1'
[stderr] if (checkSecurityPermission(env, context, (char **) authorizedAppSha1, 3)) {
[stderr] ^
[stderr] /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:19:16: warning: implicit declaration of function 'getDecryptedKey' is invalid in C99 [-Wimplicit-function-declaration]
[stderr] return getDecryptedKey(env, liteInitGPCParams, sizeof(liteInitGPCParams));
[stderr] ^
[stderr] /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:19:63: error: use of undeclared identifier 'liteInitGPCParams'
[stderr] return getDecryptedKey(env, liteInitGPCParams, sizeof(liteInitGPCParams));
[stderr] ^
[stderr] /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:19:37: error: use of undeclared identifier 'liteInitGPCParams'
[stderr] return getDecryptedKey(env, liteInitGPCParams, sizeof(liteInitGPCParams));
[stderr] ^
[stderr] 1 warning and 3 errors generated.
[stderr] [2/3] Building C object CMakeFiles/keys.dir/validation.c.o
[stderr] ninja: build stopped: subcommand failed.
[stderr] C++ build system [build] failed while executing:
[stderr] /home/expo/Android/Sdk/cmake/3.18.1/bin/ninja \
[stderr] -C \
[stderr] /home/expo/workingdir/build/packages/app/android/lib-keys-secret/.cxx/RelWithDebInfo/133g4p6s/arm64-v8a \
[stderr] keys
[stderr]     from /home/expo/workingdir/build/packages/app/android/lib-keys-secret
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] 2: Task failed with an exception.
[stderr] -----------
[stderr] * What went wrong:
[stderr] java.lang.StackOverflowError (no error message)
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] * Get more help at https://help.gradle.org
[stderr] BUILD FAILED in 13m 57s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings
1529 actionable tasks: 1529 executed
Error: Gradle build failed with unknown error. See logs for the "Run gradlew" phase for more information.`

Notice the error happenes in `Execution failed for task ':lib-keys-secret:buildCMakeRelWithDebInfo[arm64-v8a]'.`

Is lib-keys-secret folder has something to do with this error ?

@sunnylqm 
## The same error happened

```
> Task :expo-av:compileReleaseLibraryResources
> Task :expo-barcode-scanner:compileReleaseLibraryResources
> Task :expo-camera:compileReleaseLibraryResources
> Task :expo-clipboard:compileReleaseLibraryResources
> Task :expo-constants:compileReleaseLibraryResources
> Task :expo-device:compileReleaseLibraryResources
> Task :react-native-v8:externalNativeBuildRelease
> Task :expo-file-system:compileReleaseLibraryResources
> Task :expo-font:compileReleaseLibraryResources
> Task :expo-haptics:compileReleaseLibraryResources
> Task :expo-image-loader:compileReleaseLibraryResources
> Task :expo-image-manipulator:compileReleaseLibraryResources
> Task :expo-image-picker:compileReleaseLibraryResources
> Task :expo-intent-launcher:compileReleaseLibraryResources
> Task :lib-keys-secret:buildCMakeRelWithDebInfo[arm64-v8a] FAILED
> Task :expo-keep-awake:compileReleaseLibraryResources
> Task :expo-linear-gradient:compileReleaseLibraryResources
> Task :app:bundleReleaseJsAndAssets
warning: the transform cache was reset.
                    Welcome to Metro!
              Fast - Scalable - Integrated
> Task :react-native-v8:mergeReleaseJniLibFolders
> Task :app:bundleReleaseJsAndAssets
info Writing bundle output to:, /home/expo/workingdir/build/packages/app/android/app/build/generated/assets/react/release/index.android.bundle
info Writing sourcemap output to:, /home/expo/workingdir/build/packages/app/android/app/build/generated/sourcemaps/react/release/index.android.bundle.map
info Done writing bundle output
info Done writing sourcemap output
info Copying 175 asset files
info Done copying assets
[stderr] FAILURE: Build completed with 2 failures.
[stderr] 1:
[stderr] Task failed with an exception.
[stderr] -----------
[stderr] * What went wrong:
[stderr] Execution failed for task ':lib-keys-secret:buildCMakeRelWithDebInfo[arm64-v8a]'.
[stderr] >
[stderr] com.android.ide.common.process.ProcessException: ninja: Entering directory `/home/expo/workingdir/build/packages/app/android/lib-keys-secret/.cxx/RelWithDebInfo/133g4p6s/arm64-v8a'
[stderr] [1/3] Building C object CMakeFiles/keys.dir/keys.c.o
[stderr] FAILED: CMakeFiles/keys.dir/keys.c.o
[stderr] /home/expo/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --target=aarch64-none-linux-android23 --gcc-toolchain=/home/expo/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/linux-x86_64 --sysroot=/home/expo/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/linux-x86_64/sysroot -Dkeys_EXPORTS  -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security  -O2 -g -DNDEBUG -fPIC -fvisibility=hidden -MD -MT CMakeFiles/keys.dir/keys.c.o -MF CMakeFiles/keys.dir/keys.c.o.d -o CMakeFiles/keys.dir/keys.c.o -c /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c
[stderr] /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:18:57: error: use of undeclared identifier 'authorizedAppSha1'
[stderr] if (checkSecurityPermission(env, context, (char **) authorizedAppSha1, 3)) {
[stderr] ^
[stderr] /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:19:16: warning: implicit declaration of function 'getDecryptedKey' is invalid in C99 [-Wimplicit-function-declaration]
[stderr] return getDecryptedKey(env, liteInitGPCParams, sizeof(liteInitGPCParams));
[stderr] ^
[stderr] /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:19:63: error: use of undeclared identifier 'liteInitGPCParams'
[stderr] return getDecryptedKey(env, liteInitGPCParams, sizeof(liteInitGPCParams));
[stderr] ^
[stderr] /home/expo/workingdir/build/packages/app/android/lib-keys-secret/src/main/cpp/keys.c:19:37: error: use of undeclared identifier 'liteInitGPCParams'
[stderr] return getDecryptedKey(env, liteInitGPCParams, sizeof(liteInitGPCParams));
[stderr] ^
[stderr] 1 warning and 3 errors generated.
[stderr] [2/3] Building C object CMakeFiles/keys.dir/validation.c.o
[stderr] ninja: build stopped: subcommand failed.
[stderr] C++ build system [build] failed while executing:
[stderr] /home/expo/Android/Sdk/cmake/3.18.1/bin/ninja \
[stderr] -C \
[stderr] /home/expo/workingdir/build/packages/app/android/lib-keys-secret/.cxx/RelWithDebInfo/133g4p6s/arm64-v8a \
[stderr] keys
[stderr]     from /home/expo/workingdir/build/packages/app/android/lib-keys-secret
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] 2: Task failed with an exception.
[stderr] -----------
[stderr] * What went wrong:
[stderr] java.lang.StackOverflowError (no error message)
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] * Get more help at https://help.gradle.org
[stderr] BUILD FAILED in 13m 57s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings
1529 actionable tasks: 1529 executed
Error: Gradle build failed with unknown error. See logs for the "Run gradlew" phase for more information.
```

Notice the error happenes in `Execution failed for task ':lib-keys-secret:buildCMakeRelWithDebInfo[arm64-v8a]'.`

Is lib-keys-secret folder has something to do with this error ?

--- a/README.md
+++ b/README.md
@@ -68,13 +68,13 @@ Execute the following commands in the root directory to develop different busine
 
 Execute the following commands in the root directory and build target for production. Make sure each platform starts correctly and environment variables are configured correctly. For expo build please read this [doc](https://docs.expo.dev/build/setup/) to **complete some prerequisites first**.
 
-- web: `cd packages/web && yarn build`, build the static files at packages/web/web-build, for production build, see [release-web.yml](./.github/workflows/release-web.yml) job for detail.
-- ios: use expo server to build, see [release-ios.yml](./.github/workflows/release-ios.yml) job for detail.
+- web: `cd packages/web && yarn build`, build the static files at packages/web/web-build, for production build, see [release-web.yml](./.github/workflows/release-web.yml) job for details.
+- ios: use expo server to build, see [release-ios.yml](./.github/workflows/release-ios.yml) job for details.
 - android:  
-  - use expo server to build, see [release-android.yml](./.github/workflows/release-android.yml) job for detail.
-  - or use `cd packages/app/android && ENTRY_FILE=__generated__/AppEntry.js ./gradlew aR` to build in local.
-- desktop: : `cd packages/desktop && yarn build`, see [release-desktop.yml](./.github/workflows/release-desktop.yml) job for detail.
-- ext: `cd packages/ext && yarn build:all`, see [release-ext.yml](./.github/workflows/release-ext.yml) job for detail.
+  - use expo server to build, see [release-android.yml](./.github/workflows/release-android.yml) job for details.
+  - or use `cd packages/app/android && ./gradlew aR` to build in local.
+- desktop: : `cd packages/desktop && yarn build`, see [release-desktop.yml](./.github/workflows/release-desktop.yml) job for details.
+- ext: `cd packages/ext && yarn build:all`, see [release-ext.yml](./.github/workflows/release-ext.yml) job for details.
 
 ## 🗂 Multi-repository directory structure
 

--- a/packages/app/index.js
+++ b/packages/app/index.js
@@ -1,0 +1,1 @@
+import './__generated__/AppEntry.js';

